### PR TITLE
Update recipe and make run_constrained lower bound glibc.

### PR DESCRIPTION
**Pull request by b-kloss**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

This PR makes the runtime constraint on glibc version a lower bound instead of fixing it to `glibc2.17`. It also updates the recipe to the cuda versions currently supported by CI.

### CODE REVIEW

**Comment by b-kloss** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -57,15 +58,15 @@ requirements:
...
-    - {{ pin_compatible("cuda-version", max_pin="x") }}
-    - cudatoolkit >=11.1,<12  # [(cuda_compiler_version or "").startswith("11")]
-    - libcublas               # [(cuda_compiler_version or "").startswith("12")]
+    - {{ pin_compatible("cuda-version", max_pin="x", min_pin="x") }}
+    - cudatoolkit  # [(cuda_compiler_version or "").startswith("11")]
+    - libcublas    # [(cuda_compiler_version or "").startswith("12")]
   run_constrained:
     - arm-variant * {{ arm_variant_type }}  # [aarch64]
-    - {{ stdlib('c') }}
+    - __glibc >=2.17,<3.0.a0
```
Comment (carterbox) [inline recipe/meta.yaml:69]: The stdlib package already exports a dependency on __glibc when stdlib is above baseline (the first version offered on the channel for the architecture).

https://conda-metadata-app.streamlit.app/?q=conda-forge%2Fnoarch%2Fsysroot_linux-64-2.17-h4a8ded7_18.conda

Comment (b-kloss) [reply recipe/meta.yaml:69]: Could it be that the `run_constrained` constraint that gets generated on sysroot_linux is overruling the more permissive constraint in the `run` section:

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=999334&view=logs&j=4f922444-fdfe-5dcf-b824-02f86439ef14&t=b2a8456a-fb11-5506-ca32-5ccd32538dc0&l=691

Comment (b-kloss) [reply recipe/meta.yaml:69]: In that case, we should be able to remove the run_constrained constraint completely here, right?

Comment (carterbox) [reply recipe/meta.yaml:69]: I checked staged recipes for [the original version of this recipe](https://github.com/conda-forge/staged-recipes/pull/13576/files). It seems that it was merged before the `{{ stdlib('c') }}` template existed which is why it needed to manually constrain the `__glibc` package. I think this recipe was just incorrectly transitioned to use `{{ stdlib('c') }}` It should not have any explicit constraints on `__glibc` and just use the `{{ stdlibc('c') }}` package in the build requirements. I'm going to open a new PR to fix this and other open issues.

**Pull request closed without merging**